geomstats/_backend/numpy/__init__.py

```diff
@@ -246,6 +246,11 @@ def diag(x):
     return result
 
 
+def set_diag(x, new_diag):
```

Comment:
Is there a difference with this numpy function, and if not, should we call it fill_diagonal?
https://docs.scipy.org/doc/numpy/reference/generated/numpy.fill_diagonal.html

Reply:
Yes, there is a difference, because `fill_diagonal` fills indices `[i, ..., i]`, that is, the diagonal across all dimensions, while this function only wants to set the diagonal of the last two dimensions.